### PR TITLE
Update app compat libraries and usage.

### PR DESCRIPTION
**Pull request by jingibus**

### CODE REVIEW

File: criminalintent-sample/build.gradle [hunk 1]
```diff
@@ -19,8 +19,8 @@ android {
 }
 
 dependencies {
-    compile 'com.android.support:appcompat-v7:21.0.0'
-    compile 'com.android.support:recyclerview-v7:+'
+    compile 'com.android.support:appcompat-v7:22.1.1'
```
Comment (bolot) [inline criminalintent-sample/build.gradle]: why not 22.2.0?

**Comment by cstew** [comment]
Consider removing your res/values-v21 folder. You don't need to do any kind of theme resource qualification with app compat.

File: criminalintent-sample/build.gradle (hunk 1 above)
Comment (jingibus) [inline criminalintent-sample/build.gradle]: Because my support repo is out of date! Thanks.

**Comment by paul-turner** [comment]
From sample app options menu:
```
java.lang.ClassCastException: com.bignerdranch.android.criminalintent.CrimeListActivity cannot be cast to android.support.v7.app.ActionBarActivity
            at com.bignerdranch.android.criminalintent.BaseFragment.getActionBar(BaseFragment.java:10)
            at com.bignerdranch.android.criminalintent.CrimeListFragment.onOptionsItemSelected(CrimeListFragment.java:189)
            at android.app.FragmentManagerImpl.dispatchOptionsItemSelected(FragmentManager.java:1913)
            at android.app.Activity.onMenuItemSelected(Activity.java:2537)
            at android.support.v4.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:353)
            at android.support.v7.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:144)
            at android.support.v7.internal.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:99)
            at android.support.v7.app.AppCompatDelegateImplV7.onMenuItemSelected(AppCompatDelegateImplV7.java:538)
            at android.support.v7.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:802)
            at android.support.v7.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:153)
            at android.support.v7.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:949)
            at android.support.v7.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:939)
            at android.support.v7.internal.view.menu.MenuPopupHelper.onItemClick(MenuPopupHelper.java:187)
            at android.widget.AdapterView.performItemClick(AdapterView.java:298)
            at android.widget.AbsListView.performItemClick(AbsListView.java:1086)
            at android.widget.AbsListView$PerformClick.run(AbsListView.java:2859)
            at android.widget.AbsListView$1.run(AbsListView.java:3533)
            at android.os.Handler.handleCallback(Handler.java:615)
            at android.os.Handler.dispatchMessage(Handler.java:92)
            at android.os.Looper.loop(Looper.java:137)
            at android.app.ActivityThread.main(ActivityThread.java:4745)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:511)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
            at dalvik.system.NativeStart.main(Native Method)
```

**Comment by jingibus** [comment]
Ready for review

**Pull request merged**